sdk/core/core/src/az_json_parser.c

```diff
@@ -73,7 +63,7 @@ az_json_parser_stack_last(az_json_parser const* json_parser)
 AZ_NODISCARD AZ_INLINE az_result
 az_json_parser_push_stack(az_json_parser* json_parser, _az_json_stack json_stack)
```

Comment:
pushing/popping and everything related to the internal parser state should be internal.

Reply:
I agree. Deferring as outside the scope of this PR. I left the parser code unchanged to make changes in reasonable chunks.